modules/create_timepoint/php/create_timepoint.class.inc

```diff
@@ -202,10 +203,23 @@ class Create_Timepoint extends \NDB_Form
                 $this->addSelect('visitLabel', 'Visit label', $labelOptions);
                 $this->addRule('visitLabel', 'Visit label is required', 'required');
 
+                $languages = \Utility::getLanguageList();
+                array_unshift($languages, null);
+                $this->addSelect(
+                    'languageID',
+                    'Language',
+                    $languages
+                );
+                $this->addRule(
+                    'languageID',
+                    'Language is required',
+                    'required'
+                );
```

Comment:
I don't think it makes sense to do this unless `count($languages)` is `> 1`.. it doesn't make sense to require language be a required user input on unilingual projects.

Reply:
I think this ties back to https://github.com/aces/Loris/issues/5380

Reply:
if that is feasible then we can just leave it as it is

Reply:
I don't think sites and languages are comparable in that regard. Languages are a server configuration that aren't tied to the data, not having them defined should be equivalent to be a server configuration saying "this project doesn't use languages" and is just confusing to end users.. sites are variable-number-of-associations within a single loris instance.

Reply:
same applies to site, a loris instance with a single site shouldnt worry about seeing the site... it's a given that any data entered is at that site...

language in this context is tied to the data !! it's the language of collection its a parameter like any other

Reply:
If you mean "a project with only 1 psc in the table" I agree, the ticket you referenced is about dealing with users that have multiple vs 1 site the same in a study that has multiple sites.

Reply:
I leave language as required but make it default to the study language if count = 1?

Reply:
No, language should not be required or displayed if there is only 1 language configured.

Reply:
I agree with @driusan 
We should always aim at minimizing user clicks count. If there is only one language, the frontend should hide the select element. Then, having it optional or required is a matter of: do we want to rely on the default value of the column in the database.
If the field is optional, createTimepoint should default to the column default.

Reply:
@driusan I mean absolutely any drop-down with a single option. regardless of what it is...

I don't like hiding stuff for several reasons:
1. when things are hidden they don't get tested during manual testing rounds because testers omit or forget to change the necessary configurations to enable/disable a hidden field, as demonstrated with the latest bug that Tomas caught (#6008) and so many before 
2. It's also confusing for users, and user manuals where we constantly have to mention "if you have more than one site affiliation, make sure to choose a site..... If you have a multi lingual study make sure to choose a language" and even at that we get "bug reports" with title, CANNOT CHOOSE SITE OF....
3. it's inconsistent across modules and it's inconsistent with instruments
4. makes the code ugly and bug prone weather it's in the php code, the templates or in the react.

Solution, either implementing #5380 ! I think it's the concern of the view to manage "clicks" I think it should just auto select the first element when there is only one AND it is required because there is also the case where a selection is not required and then you have to display the field anyways

Reply:
Can you name a single site anywhere on the internet that forces you to choose your language with only 1 option when submitting a form?

Reply:
yes

![Screenshot from 2020-02-06 10-50-45](https://user-images.githubusercontent.com/15268287/73953562-98aa9380-48ce-11ea-8294-8123cb2ad9ce.png)

only master branch is available, it's preselected BUT THEY DON'T HIDE THE DROPDOWN !!

Reply:
since when are branches languages?

Reply:
@driusan 
![22b79a640bc5089497a6262901b7274c](https://user-images.githubusercontent.com/15268287/73954251-8c730600-48cf-11ea-855f-3193a4fafcc7.jpg)

Reply:
It's not a question of "should we display dropdowns with 1 option". That's already resolved: we display the same form to the users on the LORIS instance regardless of multi-site affiliations in order to have all users have the same experience and avoid confusion. It's a question of "should we display dropdowns with 1 option *when there is no possibility of any users on that LORIS instance ever having multiple options in that dropdown*".

In your GitHub screenshot, someone can create a branch on that repo tomorrow (in fact, if they wanted to do it through the GitHub UI they'd have to go through that dropdown) and there are other repos on GitHub which have multiple branches.

Reply:
and loris admins can add a language too... anyways, this is going too far. I made my point

Reply:
An admin adding a language would be analogous to an admin going to the configuration module and toggling the "useFeatureX"  config setting. If that were to happen, the dropdown should appear for all users.

Reply:
Okay so both options are valid. 

I think there isn't any harm in having the dropdown visible for everyone, with a default selected when only one language is available. It gives users a unified experience and also signals that there is the possibility of adding more languages (in case users or admins were not aware). 

Another reason why I am for having it displayed is because currently the integration testing we have does not cover the case where there are multiple languages (nor the case of multiple sites / projects, for that matter). This leads to the neglect of this feature during testing since the dropdown would never be displayed when using current instances of raisinbread. 

@driusan what do you think?

Reply:
No, if there is only 1 language configured for the LORIS instance there should not be a dropdown displayed.

Reply:
ok, implemented. Please review